onnxruntime/core/providers/cpu/tensor/utils.h

```diff
@@ -34,7 +34,7 @@ struct TensorPitches : std::vector<int64_t> {
 
     if (padded_rank >= 1) {
       for (size_t i = 0; i < padded_rank; ++i) {
-        if (i == 0)
+        if (i == 0 && tensor_rank > 0)  // For scalar tensor, the values in the pitches are all 1.
```

Comment:
>scalar tensor [](start = 47, length = 13)

scalar could have two shape: [1] or []

tensor_rank = 0 could also be a scalor

does this check exclude the 0 rank case ?

Reply:
The added code is actually to handle '[]' case. The previous code might cause segment fault if tensor_rank ==0 when accessing 'dim[0]' in this if clause. So the added code is to always use the else clause if tensor_rank == 0.